miden-crypto/src/word.rs

```diff
@@ -0,0 +1,614 @@
+    fn from(value: &Word) -> Self {
+        (*value).into()
+    }
+}
+
+impl From<Word> for [u8; WORD_SIZE_BYTES] {
+    fn from(value: Word) -> Self {
+        value.as_bytes()
+    }
+}
+
+impl From<&Word> for String {
+    /// The returned string starts with `0x`.
+    fn from(value: &Word) -> Self {
+        (*value).into()
+    }
+}
+
+impl From<Word> for String {
+    /// The returned string starts with `0x`.
+    fn from(value: Word) -> Self {
+        value.to_hex()
+    }
+}
+
+// CONVERSIONS: TO WORD
+// ================================================================================================
+
+impl From<&[bool; WORD_SIZE_FELT]> for Word {
+    fn from(value: &[bool; WORD_SIZE_FELT]) -> Self {
+        (*value).into()
+    }
+}
+
+impl From<[bool; WORD_SIZE_FELT]> for Word {
+    fn from(value: [bool; WORD_SIZE_FELT]) -> Self {
+        [value[0] as u32, value[1] as u32, value[2] as u32, value[3] as u32].into()
+    }
+}
+
+impl From<&[u8; WORD_SIZE_FELT]> for Word {
+    fn from(value: &[u8; WORD_SIZE_FELT]) -> Self {
+        (*value).into()
+    }
+}
+
+impl From<[u8; WORD_SIZE_FELT]> for Word {
+    fn from(value: [u8; WORD_SIZE_FELT]) -> Self {
+        Self([value[0].into(), value[1].into(), value[2].into(), value[3].into()])
+    }
+}
+
+impl From<&[u16; WORD_SIZE_FELT]> for Word {
+    fn from(value: &[u16; WORD_SIZE_FELT]) -> Self {
+        (*value).into()
+    }
+}
+
+impl From<[u16; WORD_SIZE_FELT]> for Word {
+    fn from(value: [u16; WORD_SIZE_FELT]) -> Self {
+        Self([value[0].into(), value[1].into(), value[2].into(), value[3].into()])
+    }
+}
+
+impl From<&[u32; WORD_SIZE_FELT]> for Word {
+    fn from(value: &[u32; WORD_SIZE_FELT]) -> Self {
+        (*value).into()
+    }
+}
+
+impl From<[u32; WORD_SIZE_FELT]> for Word {
+    fn from(value: [u32; WORD_SIZE_FELT]) -> Self {
+        Self([value[0].into(), value[1].into(), value[2].into(), value[3].into()])
+    }
+}
+
+impl TryFrom<&[u64; WORD_SIZE_FELT]> for Word {
+    type Error = WordError;
+
+    fn try_from(value: &[u64; WORD_SIZE_FELT]) -> Result<Self, WordError> {
+        (*value).try_into()
+    }
+}
+
+impl TryFrom<[u64; WORD_SIZE_FELT]> for Word {
+    type Error = WordError;
+
+    fn try_from(value: [u64; WORD_SIZE_FELT]) -> Result<Self, WordError> {
+        Ok(Self([
+            value[0].try_into().map_err(WordError::InvalidFieldElement)?,
+            value[1].try_into().map_err(WordError::InvalidFieldElement)?,
+            value[2].try_into().map_err(WordError::InvalidFieldElement)?,
+            value[3].try_into().map_err(WordError::InvalidFieldElement)?,
+        ]))
+    }
+}
+
+impl From<&[Felt; WORD_SIZE_FELT]> for Word {
+    fn from(value: &[Felt; WORD_SIZE_FELT]) -> Self {
+        Self(*value)
+    }
+}
+
+impl From<[Felt; WORD_SIZE_FELT]> for Word {
+    fn from(value: [Felt; WORD_SIZE_FELT]) -> Self {
+        Self(value)
+    }
+}
+
+impl TryFrom<&[u8; WORD_SIZE_BYTES]> for Word {
+    type Error = HexParseError;
```

Comment:
Using `HexParserError` here feels a bit off (since we are not parsing a hex string). Should it be just `WordError` here?